frontend/src/Context/SessionProvider/session.tsx

```diff
@@ -186,6 +193,39 @@ const SessionProvider = ({ children }: SessionProviderProps): ReactElement => {
     });
   }, []);
 
+  /**
+   * isMaxPinned {boolean} - whether the maximum number of allowed pinned participants has been reached.
+   * This is used to disable the pin buttons so no more participants can be pinned.
+   */
+  const isMaxPinned = useMemo(() => {
+    const pinnedCount = subscriberWrappers.filter((sub) => sub.isPinned).length;
+    return pinnedCount >= MAX_PIN_COUNT;
+  }, [subscriberWrappers]);
+
+  /**
+   * Marks a subscriber as pinned, and moves it to the top of the display order.
+   * @param {string} id - The ID of the subscriber to pin.
+   */
+  const pinSubscriber = useCallback(
+    (id: string) => {
+      setSubscriberWrappers((previousSubscriberWrappers) => {
+        const subscribers = previousSubscriberWrappers
+          .map((subscriberWrapper) => {
+            if (subscriberWrapper.id === id) {
+              return {
+                ...subscriberWrapper,
+                isPinned: !subscriberWrapper.isPinned,
+              };
+            }
+            return subscriberWrapper;
+          })
+          .sort(sortByDisplayPriority(activeSpeakerId)); // Sorting by display priority will place this pinned participant above the rest
+        return subscribers;
```

Comment:
Could we break the state function into a helper? I know we plan on refactoring later, but it might make sense to separate this logic so we can add the test now. Hopefully, the tests we write for this helper, could still be reused.

Reply:
I see what you're saying, let me give that a whirl

Reply:
done